### PR TITLE
NXP-30529: show send to coldstorage btn if rendition available

### PR DESCRIPTION
**Pull request by andreacornaglia**

### CODE REVIEW

File: nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html [hunk 1]
```diff
@@ -77,9 +77,18 @@ <h2>[[i18n('documentContentView.moveToColdStorage.ask.confirmation')]]</h2>
           !this.hasFacet(document, 'ColdStorage')
       && (this._hasAdministrationPermissions(currentUser) || this.hasPermission(document, 'WriteColdStorage'))
       && this.hasContent(document)
+      && this._isRenditionAvailable(document)
         );
       }
 
+      _isRenditionAvailable(document) {
+        if(document.type === 'Picture'){
```
Comment (mnixo) [inline nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html]:
```suggestion
        if (document.type === 'Picture') {
```

**Review by mnixo — DISMISSED** [review]

File: nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html [hunk 2]
```diff
@@ -77,9 +77,18 @@ <h2>[[i18n('documentContentView.moveToColdStorage.ask.confirmation')]]</h2>
...
+          return document.properties['picture:views'][0].filename !== 'empty_picture.png';
+        } else if(document.type === 'Video') {
```
Comment (mnixo) [inline nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html]:
```suggestion
        } else if (document.type === 'Video') {
```

Comment (mnixo) [inline nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html]:
```suggestion
        } else if (document.type === 'Video') {
```

**Review by mnixo — DISMISSED** [review]

**Review by mnixo — DISMISSED** [review]

File: nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html [hunk 3]
```diff
@@ -77,9 +77,20 @@ <h2>[[i18n('documentContentView.moveToColdStorage.ask.confirmation')]]</h2>
           !this.hasFacet(document, 'ColdStorage')
       && (this._hasAdministrationPermissions(currentUser) || this.hasPermission(document, 'WriteColdStorage'))
       && this.hasContent(document)
+      && this._isRenditionAvailable(document)
         );
       }
 
+      _isRenditionAvailable(document) {
+        let result = true;
+        if (document.type === 'Picture') {
```
Comment (Gabez0r) [inline nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html]: Any particular reason for not checking instead if the document has facet "Picture" or "Video"?

Comment (mnixo) [reply nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html]: That's a good point, checking for facets ensures that other doctypes extending `Picture` and `Video` would also benefit from this addition.

**Review by doulba — DISMISSED** [review]

File: nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html (hunk 3 above)
Comment (andreacornaglia) [reply nuxeo-coldstorage-web/src/main/resources/web/nuxeo.war/ui/coldstorage/nuxeo-move-content-to-coldstorage-button.html]: Good point! I forgot about facets. chaning the code, thanks for the feedback!

**Review by Gabez0r — DISMISSED** [review]

**Review by guirenard — DISMISSED** [review]

**Review by mnixo — DISMISSED** [review]

**Review by mnixo — APPROVED** [review]

**Review by guirenard — APPROVED** [review]

**Pull request merged**